### PR TITLE
feat(github-bot): add a fork condition and handle PR reviews

### PR DESCRIPTION
**Pull request by aeddi**
This PR improves the bot on two points:
- it now handle `pull_request_review` events to address [this concern](https://github.com/gnolang/gno/issues/3238#issuecomment-2526206058) https://github.com/gnolang/gno/commit/4f7b0b80c6e726806a473556b02444fded707254
- a new condition allows to check if a PR was created from a fork to address [this concern](https://github.com/gnolang/gno/issues/3238#issuecomment-2524018469) https://github.com/gnolang/gno/commit/f491d95d68c755d7154cbbee48a9cebc7693fa89

### CODE REVIEW

**Comment by Gno2D2** [comment]
I'm a bot that assists the Gno Core team in maintaining this repository. My role is to ensure that contributors understand and follow our guidelines, helping to streamline the development process.

The following requirements must be fulfilled before a pull request can be merged.
Some requirement checks are automated and can be verified by the CI, while others need manual verification by a staff member.

These requirements are defined in this [configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

## Automated Checks

 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🔴 The pull request head branch must be up-to-date with its base ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch))


## Manual Checks

*No manual checks match this pull request.*

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>The pull request head branch must be up-to-date with its base (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 Head branch (aeddi:dev/aeddi/github-bot-improvements) is up to date with base (master): behind by 3 / ahead by 2

```
</blockquote></details>

</blockquote></details>



</blockquote></details>

**Review by thehowl — APPROVED** [review]
noice

**Pull request merged**
